python-avd/pyavd/_schema/avdvalidator.py

```diff
@@ -58,7 +58,7 @@ def validate(self, instance: Any, schema: dict | None = None, path: list[str | i
 
     def type_validator(self, schema_type: str, instance: Any, _schema: dict, path: list[str | int]) -> Generator:
         """Validates the type of `instance` equal to `schema_type`."""
-        if not is_type(instance, schema_type):
+        if not is_type(instance, cast("Literal['dict', 'int', 'str', 'bool', 'list']", schema_type)):
```

Comment:
why cast here? you should change the funciton signature to take a Literal or change `is_type` to take a `str` rather than a Literal for `type_str`

Reply:
Fixed